libs/dtm0/dtm0_lib.py

```diff
@@ -0,0 +1,159 @@
+                                   validate_certs=S3_CFG["validate_certs"])
+            self.log.info("Workload: %s objects of %s with %s parallel clients.",
+                          samples, size, clients)
+            self.log.info("Log Path %s", resp[1])
+            log_path = resp[1]
+            if s3bench.check_log_file_error(resp[1]):
+                results.append(False)
+                break
+            else:
+                results.append(True)
+                workload.append({'bucket': bucket_name, 'obj_name_pref': object_name,
+                                 'num_clients': clients, 'obj_size': size,
+                                 'num_sample': samples})
+        if all(results):
+            queue.put([True, workload])
+        else:
+            queue.put([False, f"S3bench workload for failed."
+                             f" Please read log file {log_path}"])
+
+    def perform_read_op(self, workload_info: list, queue, skipread: bool = True,
+                        validate: bool = True, skipcleanup: bool = False, loop=1):
+        """
+        Perform read operations
+        :param workload_info: List Workload to read/validate/delete
+        :param queue: Multiprocessing Queue to be used for returning values (Boolean,dict)
+        :param skipread: Skip read
+        :param validate: Validate checksum
+        :param skipcleanup: Skip Cleanup
+        :param loop: Loop count for performing reads in iteration.
+        """
+        results = list()
+        log_path = None
+        for i in range(loop):
+            for workload in workload_info:
+                resp = s3bench.s3bench(self.access_key,
+                                       self.secret_key,
+                                       bucket=workload['bucket'],
+                                       num_clients=workload['num_clients'],
+                                       num_sample=workload['num_sample'],
+                                       obj_name_pref=workload['obj_name_pref'],
+                                       obj_size=workload['obj_size'],
+                                       skip_cleanup=skipcleanup,
+                                       skip_write=True,
+                                       skip_read=skipread,
+                                       validate=validate,
+                                       log_file_prefix=f"read_workload_{workload['obj_size']}mb",
+                                       end_point=S3_CFG["s3_url"],
+                                       validate_certs=S3_CFG["validate_certs"])
+                self.log.info(f"Workload: %s objects of %s with %s parallel clients ",
+                              workload['num_sample'], workload['obj_size'],
+                              workload['num_clients'])
+                self.log.info(f"Log Path {resp[1]}")
+                log_path = resp[1]
+                if s3bench.check_log_file_error(resp[1]):
```

Comment:
I think you need to use system_utils.validate_s3bench_parallel_execution for log verification, as this method handles all the errors.

Reply:
As discussed the requirements of both the functions are different. keeping the code as is. 
cc @sonalk0209 @seagate-sarang-sawant